index.js

```diff
@@ -55,6 +62,11 @@
   }
   extend(Nothing, Maybe);
```

Comment:
I'd like to have just a single value of the Nothing type, but I haven't worked out a solution which meets all of the following requirements:

  - `new S.Nothing()` must return the value;
  - `S.Nothing()` must return the value;
  - applying `R.is(S.Nothing)` to the value must evaluate true; and
  - the `name` property of the value's constructor must be `'Nothing'`.

In fact, I don't think it's possible to meet all these requirements.